From 10d2a977ee1d469f0bf9059bb2d0b55fd2eecbac Mon Sep 17 00:00:00 2001
From: Andre McCurdy <armccurdy@gmail.com>
Date: Wed, 20 Jan 2016 13:00:00 -0800
Subject: [PATCH] make gio_unix_2_0 dependency configurable

Prior to 1.7.1, gst-plugins-base accepted a configure option to
disable gio_unix_2_0, however it was implemented incorrectly using
AG_GST_CHECK_FEATURE. That was fixed in 1.7.1 by making the
dependency unconditional.

  http://cgit.freedesktop.org/gstreamer/gst-plugins-base/commit/?id=aadefefba88afe4acbe64454650f24e7ce7c8d70

To make builds deterministic, re-instate support for
--disable-gio_unix_2_0, but implement it using the AC_ARG_ENABLE
instead of AG_GST_CHECK_FEATURE.

Upstream-Status: Pending

Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 configure.ac | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/configure.ac b/configure.ac
index 9c52aeb..26cacd6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -747,9 +747,16 @@ AC_SUBST(FT2_CFLAGS)
 AC_SUBST(FT2_LIBS)
 
 dnl *** gio-unix-2.0 for tests/check/pipelines/tcp.c ***
+AC_ARG_ENABLE([gio_unix_2_0],
+  [AS_HELP_STRING([--disable-gio_unix_2_0],[disable use of gio_unix_2_0])],
+  [],
+  [enable_gio_unix_2_0=yes])
+
+if test "x${enable_gio_unix_2_0}" != "xno"; then
 PKG_CHECK_MODULES(GIO_UNIX_2_0, gio-unix-2.0 >= 2.24,
     HAVE_GIO_UNIX_2_0="yes",
     HAVE_GIO_UNIX_2_0="no")
+fi
 AM_CONDITIONAL(USE_GIO_UNIX_2_0, test "x$HAVE_GIO_UNIX_2_0" = "xyes")
 
 dnl *** finalize CFLAGS, LDFLAGS, LIBS
-- 
1.9.1

